src/vs/editor/contrib/referenceSearch/browser/referenceSearch.ts

```diff
@@ -7,7 +7,7 @@
 import * as nls from 'vs/nls';
 import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
 import URI from 'vs/base/common/uri';
-import { TPromise } from 'vs/base/common/winjs.base';
+import { PPromise, TPromise } from 'vs/base/common/winjs.base';
```

Comment:
I am passing in the PPromise to toggleWidget so it can render progress. I could have continued to propogate the callback function independenty but
1. It would have changed the function signature of toggleWidget, even for cases where progress isn't used (these exist), and
2. I would have to create some sort of intermediate object anyway to store the callback so that the implementation of toggleWidget could subscribe to listen to the callback.

Since PPromise does exactly this, and this is completely internal to VS Code, it seems appropriate to use here.